commands/files/readerfile.go

```diff
@@ -13,10 +13,12 @@ type ReaderFile struct {
 	fullpath string
 	reader   io.ReadCloser
 	stat     os.FileInfo
+	offset   int64
+	baseInfo ExtraInfo
 }
 
-func NewReaderFile(filename, path string, reader io.ReadCloser, stat os.FileInfo) *ReaderFile {
-	return &ReaderFile{filename, path, reader, stat}
+func NewReaderFile(filename, path, abspath string, reader io.ReadCloser, stat os.FileInfo) *ReaderFile {
```

Comment:
All right I will start by trying to change FullPath to an absolute path add see what happens when used with "add -r".  I am will make sure the contents of the directory objects are correct.